### PR TITLE
Added the logic to fetch member details Also in SSR

### PR DESCRIPTION
**Pull request by sumitd94**
This is in reference to issue #53 
(WIP PR)

### CODE REVIEW

File: pages/index.js [hunk 1]
```diff
@@ -19,27 +20,45 @@ export async function getServerSideProps() {
   const URL =
```
Comment (shihabus) [inline pages/index.js:20]: Thought: if you do SSR for these many users(in future), the initial load time will be too much. I think hybrid rendering is better. If SEO is a concern, render a loading view from server and fetch the data in client. 
@ankushdharkar @invalidtoken

**Review by shihabus — COMMENTED** [review]
Couldn’t go deep.

File: components/member-list-item/index.js [hunk 2]
```diff
@@ -25,34 +22,42 @@ const PreviewMember = (props) => {
...
+        {memberDetails ? `${memberDetails['first_name']} ${memberDetails['last_name']}` : id}
       </h2>
-      {data && (
+      {memberDetails && (
         <div className={classNames.iconsContainer}>
-          {data['twitter_id'] && <SocialMediaIcon id={data.twitter_id} type="twitter" />}
-          {data['github_id'] && <SocialMediaIcon id={data.github_id} type="github" />}
-          {data['linkedin_id'] && <SocialMediaIcon id={data.linkedin_id} type="linkedIn" />}
-          {data['instagram_id'] && <SocialMediaIcon id={data.instagram_id} type="instagram" />}
+          {memberDetails['twitter_id'] && (
```
[the remark this answers is not in the document]
Comment (sumitd94) [reply components/member-list-item/index.js]: I have followed a even more simpler way, could you please check that on my latest commit

File: pages/index.js [hunk 3]
```diff
@@ -19,27 +20,45 @@ export async function getServerSideProps() {
...
   try {
     const res = await fetch(URL);
     if (res.status !== 200) {
       throw new Error(
         'There was some issues fetching the members, Please try again after some time'
       );
     }
-    const data = await res.json();
-    return { props: { data } };
+    let data = await res.json();
```
Comment (ankushdharkar) [inline pages/index.js:35]: Why not const?

File: pages/index.js [hunk 4]
```diff
@@ -19,27 +20,45 @@ export async function getServerSideProps() {
   const URL =
     'https://raw.githubusercontent.com/Real-Dev-Squad/website-static/main/ids/mapping.json';
 
+  const membersArr = [];
+  let memberDetailsUrl = '';
+  let resp1 = '';
```
Comment (ankushdharkar) [inline pages/index.js:25]: Too generic of a name, can we describe more meaningfully what the purpose of this response is?

**Review by ankushdharkar — APPROVED** [review]

File: pages/index.js [hunk 5]
```diff
@@ -19,27 +20,45 @@ export async function getServerSideProps() {
...
+      if (resp1.status == 200) {
+        jsonObj = await resp1.json();
+        membersArr.push({
+          ...jsonObj,
+          img_url: getImgURL(rdsid)
+        });
+      }
+    }
+
+    return { props: { membersArr } };
```
Comment (ankushdharkar) [inline pages/index.js:49]: Can be parallelized

Comment (niyasrahman) [reply pages/index.js:49]: I was randomly checking this MR and am confused with the word parallelized. Can some explain me what is paralleliztion?

**Pull request merged**